.github/workflows/release-analysis.yml

```diff
@@ -4,31 +4,70 @@ on:
   push:
     tags: 'ana-*.*.*'
 
-jobs:
-  build_project:
-    name: Build Project and Release
+env:
+  VERSION: "0.0.0"
 
+jobs:
+  promote_and_release:
+    name: Promote and Release
     runs-on: ubuntu-latest
-
     permissions:
       contents: write
+      packages: write
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
-      - name: Setup Node.js environment
-        uses: actions/setup-node@v3
+      # Extract version from tag
+      - name: Set version
+        run: |
+          echo "VERSION=${GITHUB_REF#refs/tags/ana-}" >> $GITHUB_ENV
+
+      # Build just for release artifacts
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
         with:
           node-version: '20'
+          registry-url: 'https://registry.npmjs.org'
 
       - name: Build analysis
         working-directory: ./analysis
         run: |
           chmod +x ./gradlew
           ./gradlew build
```

Comment:
You might want to swap to this assemble, build will run all tests.
You could consider adding a test run to the visualization instead, would be fine be me, as it is a new commit after all.
Either way, we might want to disable the regular test workflow, so that the release doesn't trigger it additionally.